src/components/tl/mlx5/tl_mlx5_rcache.c

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5.h"
+
+static ucs_status_t
+rcache_reg_mr(void *context, ucs_rcache_t *rcache, //NOLINT: rcache is unused
+              void *arg, ucc_rcache_region_t *rregion,
+              uint16_t flags) //NOLINT: flags is unused
+{
+    ucc_tl_mlx5_context_t *ctx      = (ucc_tl_mlx5_context_t *)context;
+    void *                 addr     = (void *)rregion->super.start;
+    ucc_tl_mlx5_reg_t *    mlx5_reg = ucc_tl_mlx5_get_rcache_reg_data(rregion);
+    int *                  change_flag = (int *)arg;
+    size_t length = (size_t)(rregion->super.end - rregion->super.start);
+
+    mlx5_reg->region = rregion;
+    *change_flag     = 1;
```

Comment:
what is change_flag? why 1?

Reply:
This flag is set to one in case of a new buffer registration (i.e. the entry is not found in the cache).
Note that this is used inside `ucc_tl_mlx5_reg_fanin_start` when we call `ucc_rcache_get`